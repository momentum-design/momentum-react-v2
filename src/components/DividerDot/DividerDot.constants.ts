const CLASS_PREFIX = 'md-divider-dot';

const DEFAULTS = {};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
