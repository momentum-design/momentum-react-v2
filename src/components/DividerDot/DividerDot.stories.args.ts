import { commonStyles } from '../../storybook/helper.stories.argtypes';

export default {
  ...commonStyles,
};
