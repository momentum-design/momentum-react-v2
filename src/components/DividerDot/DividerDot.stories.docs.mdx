The `<DividerDot />` component provides a dot to be used as a divider between text or other content that needs to be separated.