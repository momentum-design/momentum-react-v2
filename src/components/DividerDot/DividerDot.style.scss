.md-divider-dot-wrapper {
  display: inline-block;
  width: 0.25rem;
  height: 0.25rem;
  margin: 0 0.25rem;
  vertical-align: middle;
  background-color: var(--listitem-dotdivider);
  border-radius: 50%;
}
