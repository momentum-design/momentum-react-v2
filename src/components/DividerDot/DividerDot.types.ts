import { CSSProperties } from 'react';

export interface Props {
  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;
}
