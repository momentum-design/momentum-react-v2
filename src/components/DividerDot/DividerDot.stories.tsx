import { Template } from '../../storybook/helper.stories.templates';
import { DocumentationPage } from '../../storybook/helper.stories.docs';
import StyleDocs from '../../storybook/docs.stories.style.mdx';

import DividerDot, { DividerDotProps } from './';
import argTypes from './DividerDot.stories.args';
import Documentation from './DividerDot.stories.docs.mdx';

export default {
  title: 'Momentum UI/DividerDot',
  component: DividerDot,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
  },
};

const Example = Template<DividerDotProps>(DividerDot).bind({});

Example.argTypes = { ...argTypes };

export { Example };
