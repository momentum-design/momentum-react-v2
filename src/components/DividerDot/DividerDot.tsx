import React, { FC } from 'react';
import classnames from 'classnames';

import { STYLE } from './DividerDot.constants';
import { Props } from './DividerDot.types';
import './DividerDot.style.scss';

/**
 * The DividerDot component.
 */
const DividerDot: FC<Props> = (props: Props) => {
  const { className, id, style } = props;

  return <div className={classnames(className, STYLE.wrapper)} id={id} style={style} />;
};

export default DividerDot;
