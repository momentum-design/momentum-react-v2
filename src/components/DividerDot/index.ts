import { default as DividerDot } from './DividerDot';
import * as CONSTANTS from './DividerDot.constants';
import { Props } from './DividerDot.types';

export { CONSTANTS as DIVIDER_DOT_CONSTANTS };

export type DividerDotProps = Props;

export default DividerDot;
