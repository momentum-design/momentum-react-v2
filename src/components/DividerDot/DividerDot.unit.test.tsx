import React from 'react';
import { mount } from 'enzyme';

import DividerDot, { DIVIDER_DOT_CONSTANTS as CONSTANTS } from './';

describe('<DividerDot />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(<DividerDot />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(<DividerDot className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = mount(<DividerDot id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = mount(<DividerDot style={style} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(<DividerDot />)
        .find(DividerDot)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = mount(<DividerDot className={className} />)
        .find(DividerDot)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      const element = mount(<DividerDot id={id} />)
        .find(DividerDot)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = mount(<DividerDot style={style} />)
        .find(DividerDot)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });
  });
});
