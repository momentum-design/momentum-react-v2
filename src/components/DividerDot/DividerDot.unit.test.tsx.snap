// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DividerDot /> snapshot should match snapshot 1`] = `
<DividerDot>
  <div
    className="md-divider-dot-wrapper"
  />
</DividerDot>
`;

exports[`<DividerDot /> snapshot should match snapshot with className 1`] = `
<DividerDot
  className="example-class"
>
  <div
    className="example-class md-divider-dot-wrapper"
  />
</DividerDot>
`;

exports[`<DividerDot /> snapshot should match snapshot with id 1`] = `
<DividerDot
  id="example-id"
>
  <div
    className="md-divider-dot-wrapper"
    id="example-id"
  />
</DividerDot>
`;

exports[`<DividerDot /> snapshot should match snapshot with style 1`] = `
<DividerDot
  style={
    {
      "color": "pink",
    }
  }
>
  <div
    className="md-divider-dot-wrapper"
    style={
      {
        "color": "pink",
      }
    }
  />
</DividerDot>
`;
